spec/filters/occurrences_spec.rb

```diff
@@ -0,0 +1,71 @@
+require File.join(File.dirname(__FILE__), "..", "helpers")
+require "sensu/extensions/filters/occurrences"
+
+describe "Sensu::Extension::Occurrences" do
+  include Helpers
+
+  before do
+    @extension = Sensu::Extension::Occurrences.new
+  end
+
+  it "can filter events using occurrences" do
+    async_wrapper do
+      event = event_template
+      @extension.safe_run(event) do |output, status|
+        expect(status).to eq(1)
+        event[:check][:occurrences] = 3
+        @extension.safe_run(event) do |output, status|
+          expect(status).to eq(0)
+          event[:occurrences] = 3
+          @extension.safe_run(event) do |output, status|
+            expect(status).to eq(1)
+            async_done
+          end
+        end
+      end
+    end
+  end
+
+  it "can filter events using occurrences with a refresh" do
+    async_wrapper do
+      event = event_template
+      event[:check][:interval] = 2
+      @extension.safe_run(event) do |output, status|
+        expect(status).to eq(1)
+        event[:check][:occurrences] = 3
+        @extension.safe_run(event) do |output, status|
+          expect(status).to eq(0)
+          event[:occurrences] = 3
+          @extension.safe_run(event) do |output, status|
+            expect(status).to eq(1)
+            event[:occurrences] = 9
+            @extension.safe_run(event) do |output, status|
+              expect(status).to eq(0)
+              event[:check][:refresh] = 12
```

Comment:
My understanding is that this is testing an event should not match the filter with refresh of 12, do we need a positive match case for refresh as well?